src/scheduler/pbs_sched.cpp

```diff
@@ -540,26 +540,54 @@ static void
 connect_svrpool()
 {
 	int i;
-	svr_conn_t *svr_conns;
+	svr_conn_t *svr_conns_primary = NULL;
+	svr_conn_t *svr_conns_secondary = NULL;
 	int registered = 0;
+	int num_conf_svrs = get_num_servers();
 
 
 	while (clust_primary_sock < 0 || clust_secondary_sock < 0 || registered == 0) {
 		if (clust_primary_sock < 0) {
 			clust_primary_sock = pbs_connect(NULL);
+			/* clust_primary_sock later is used to fetch primary connections for all the servers */
 			if (clust_primary_sock < 0) {
 				/* wait for 2s for not to burn too much CPU, and then retry connection */
 				sleep(2);
 				continue;
 			}
 		}
 		clust_secondary_sock = pbs_connect(NULL);
+		/* clust_secondary_sock later is used to fetch secondary connections for all the servers */
```

Comment:
Merge these two comments into 1 and put it at the top of the if() block.  Say 'pbs_connect() will return a connection handle for all servers'